### PR TITLE
[QoL] Added https and server url is read from the env now

### PR DESCRIPTION
**Pull request by CodeTappert**
Fixes https://github.com/pagefaultgames/pokerogue/issues/1763

## Why am I doing these changes?

They were requested by walker

## What did change?

Added http prefix. And server url can now be read from env

### Screenshots/Videos

N/A

## How to test the changes?

Use local server and set the env variable to the url of the server

## Checklist
- [x] There is no overlap with another PR?
- [x] The PR is self-contained and cannot be split into smaller PRs?
- [x] Have I provided a clear explanation of the changes?
- [x] Have I tested the changes (manually)?
    - [x] Are all unit tests still passing? (`npm run test`)
- [ ] Are the changes visual?
  - [ ] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by flx-sta** [comment]
Please make sure to add the env to the `vite.env.d.ts` file

**Comment by CodeTappert** [comment]
> Please make sure to add the env to the `vite.env.d.ts` file

done

**Review by flx-sta — APPROVED** [review]
Using env.development
```
VITE_BYPASS_LOGIN=1
VITE_BYPASS_TUTORIAL=1
VITE_SERVER_URL=https://api.pokerogue.net
```
image

Using env.development
```
VITE_BYPASS_LOGIN=1
VITE_BYPASS_TUTORIAL=0
VITE_SERVER_URL=http://localhost:8001
```
image

-----

Something I noticed (has nothing TODO with the PR):
When I'm using a `.env.local` file it doesn't seem to take priority over the `env.development`.
Anyone know why?

File: .env [hunk 1]
```diff
@@ -1,2 +1,3 @@
 VITE_BYPASS_LOGIN=0
-VITE_BYPASS_TUTORIAL=0
\ No newline at end of file
+VITE_BYPASS_TUTORIAL=0
+VITE_SERVER_URL=http://localhost:8001
```
Comment (ben-lear) [inline .env:3]: nit: Isn’t the default port for the application 8000?

**Comment by CodeTappert** [comment]
So like @XGSleepWalker i think this is for you to review

**Comment by flx-sta** [comment]
BTW if i had to choose I'd simplify it even more:

-----

.env
```env
VITE_BYPASS_LOGIN=0
VITE_BYPASS_TUTORIAL=0
VITE_SERVER_URL=https://api.pokerogue.net
```

-----

.env.development
```env
VITE_BYPASS_LOGIN=1
VITE_BYPASS_TUTORIAL=0
VITE_SERVER_URL=http://localhost:8001 
```

-----

`src/utils.ts`
```ts
// [...]
// vvv - no need for this magic - vvv
// export const localServerUrl = import.meta.env.VITE_SERVER_URL ?? `http://${window.location.hostname}:${window.location.port+1}`;

export const apiUrl = import.meta.env.VITE_SERVER_URL ?? "https://api.pokerogue.net";

export function setCookie(cName: string, cValue: string): void {
// [...]
```

**Review by XGSleepWalker — APPROVED** [review]

**Pull request merged**